sdk/azidentity/chained_token_credential.go

```diff
@@ -44,24 +45,30 @@ func NewChainedTokenCredential(sources []azcore.TokenCredential, options *Chaine
 // opts: Options for the token request, in particular the desired scope of the access token.
 func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (token *azcore.AccessToken, err error) {
 	var errList []CredentialUnavailableError
-	for _, cred := range c.sources {
-		token, err = cred.GetToken(ctx, opts)
-		var credErr CredentialUnavailableError
-		if errors.As(err, &credErr) {
-			errList = append(errList, credErr)
-		} else if err != nil {
-			var authFailed AuthenticationFailedError
-			if errors.As(err, &authFailed) {
-				err = fmt.Errorf("Authentication failed:\n%s\n%s"+createChainedErrorMessage(errList), err)
-				authErr := newAuthenticationFailedError(err, authFailed.RawResponse())
-				return nil, authErr
+
+	if c.successfulCredential != nil {
+		successfulCredential := c.successfulCredential
+		token, err = successfulCredential.GetToken(ctx, opts)
+		if err != nil {
+			return nil, formatError(err, errList)
+		}
+		return token, nil
+	} else {
```

Comment:
The `else` is redundant and can be removed.

Reply:
True! Thank you!